include/spdk/env.h

```diff
@@ -323,6 +335,546 @@ int spdk_pci_addr_fmt(char *bdf, size_t sz, const struct spdk_pci_addr *addr);
  */
 void *spdk_call_unaffinitized(void *cb(void *arg), void *arg);
 
+struct spdk_ring;
+
+#define RING_F_SP_ENQ 0x0001 /**< The default enqueue is "single-producer". */
+#define RING_F_SC_DEQ 0x0002 /**< The default dequeue is "single-consumer". */
+
+/**
+ * Enqueue several objects on the ring (multi-producers safe).
```

Comment:
Many of these comments look identical to the ones from DPDK's headers; if so, we should also copy the corresponding copyright information.